packages/flutter_notification_center/lib/src/services/notification_service.dart

```diff
@@ -0,0 +1,51 @@
+import "dart:async";
+
+import "package:flutter/material.dart";
+import "../models/notification.dart";
+
+/// An abstract class representing a service for managing notifications.
+abstract class NotificationService with ChangeNotifier {
+  /// Creates a new [NotificationService] instance.
+  ///
+  /// The [listOfActiveNotifications] parameter specifies the
+  ///  list of active notifications,
+  /// with a default value of an empty list.
+  ///
+  /// The [listOfPlannedNotifications] parameter specifies the
+  ///  list of planned notifications,
+  /// with a default value of an empty list.
+  NotificationService({
+    this.listOfActiveNotifications = const [],
+    this.listOfPlannedNotifications = const [],
+  });
+
+  /// A list of active notifications.
+  List<NotificationModel> listOfActiveNotifications;
+
+  /// A list of planned notifications.
+  List<NotificationModel> listOfPlannedNotifications;
+
+  /// Pushes a notification to the service.
+  Future pushNotification(NotificationModel notification);
+
+  /// Retrieves the list of active notifications.
+  Future<List<NotificationModel>> getActiveNotifications();
+
+  /// Creates a scheduled notification.
+  Future createScheduledNotification(NotificationModel notification);
+
+  /// Creates a recurring notification.
+  Future createRecurringNotification(NotificationModel notification);
+
```

Comment:
why is there no deleteRecurringNotification?

Reply:
There is no deleteRecurringNotification because a recurring notification is actually just a scheduled notification with a new instance getting created regularly. They share the same DB collection as well, thats why they can be deleted with the same function.

But I will make this more clear by renaming `deleteScheduledNotification` with a more generic name.